pkgs/tools/X11/sunpaper/default.nix

```diff
@@ -0,0 +1,55 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, sunwait
+, wallutils
+, rPackages
+, pkgs
+}:
+
+stdenv.mkDerivation rec {
+  pname = "sunpaper";
+  version = "unstable-2022-04-01";
+
+  src = fetchFromGitHub {
+    owner = "hexive";
+    repo = "sunpaper";
+    rev = "8d518dfddb5e80215ef3b884ff009df1d4bb74c2";
+    sha256 = "sCG7igD2ZwfHoRpR3Kw7dAded4hG2RbMLR/9nH+nZh8=";
+  };
+
+  buildInputs = with pkgs; [
+    wallutils
+    sunwait
+  ];
+
+  postPatch = ''
+    substituteInPlace sunpaper.sh \
+      --replace '$HOME/sunpaper/images/' "$out/images/" \
+      --replace '$HOME/.cache' "$TMPDIR/.sunpapercache" \
```

Comment:
That would disable caching. If the directory gets written in the sandbox we should set HOME to a tempdir.

Reply:
@SuperSandro2000 - Solid callout, thank you. I updated it to have the cache in the nix store but then I realized that's immutable. I'm trying to figure out best practices for a temp dir and I guess `$TMPDIR` isn't it. Do you have a line that would work here?

Reply:
I tried this locally, but if I removed line 28 and 29, it doesn't seem like the sandbox `HOME` variable would be used at runtime right? It would only be set during this build.  

Thanks for your patience working with me @SuperSandro2000. Still trying to get my head around how to do this properly for my first package :D

Reply:
`$HOME/.cache` is mostly right XDG-wise[^0] so it'd make sense to leave it as-is when not in the build?

[^0]: It's supposed to be possible to override it with an environment variable too

Reply:
Aha! I was overcomplicating things then. I'm going to let the script manage it as it currently is upstream, in `$HOME/.cache` and not replace it.